### DISCUSSION TITLE
load all collections

### DISCUSSION THREAD

**Discussion by LMERose** [milvus-io/milvus#35276] [Q&A and General discussion]
I would like to know if milvus can implement a loop to load and retrieve all collections contained in one of the libraries to the big language model? If so, how to determine the number of retrievals? Can the above function be realized?

**Comment by yhmo** [comment]
"load and retrieve all collections contained in one of the libraries to the big language model" ------------------------------- I don't understand your purpose, could you provide a concrete example?

**Reply by LMERose** [reply]
I want to konw how to load all the collections to search the top-k?

**Reply by yhmo** [reply]
Load all collections:
```
for name in utility.list_collections():
    collection = Collection(name)
    collection.load()
```

Search topk in a collection:
```
res = client.search(
    collection_name="xxx",
    data=[
        [0.19886812562848388, 0.06023560599112088, ......, 0.838729485096104],
        [0.3172005263489739, 0.9719044792798428, ......, 0.95791889146345]
    ], # Replace with your query vectors
    limit=2, # Max. number of search results to return
    search_params={"metric_type": "IP", "params": {}} # Search parameters
)
```
https://milvus.io/docs/single-vector-search.md

**Reply by LMERose** [reply]
Is loading collections this just a matter of specifying the library?Is it possible to automatically list the collections in the library directly?If I want to combine it with langchain for knowledge base quizzing, can this feature be realized?

**Reply by yhmo** [reply]
"library" ---------- what does this mean? Milvus has no such concept. 
A milvus server has multiple databases: https://milvus.io/docs/manage_databases.md#Manage-Databases; one database has multiple collections; one collection has multiple partitions; one partition has multiple segments.

In langchain source code, milvus python sdk is encapsulated by langchain VectorStore class. langchain VectorStore has its interfaces to define operations. I didn't see any interface to load all collection in langchian. You can call milvus python sdk separately to load all collections.
